communities/src/migrations.rs

```diff
@@ -103,43 +220,70 @@ pub mod v1 {
 
 			// For community metadata, we do not need any checks, because the data is bounded already due to the CommmunityMetadata validate() function.
 
-			Ok((cid_count, cids_by_geohash_count, locations_by_geohash_count, bootstrappers_count)
+			Ok((
+				cid_count,
+				cmeta_count,
+				cids_by_geohash_count,
+				locations_by_geohash_count,
+				bootstrappers_count,
+			)
 				.encode())
 		}
 
+		/// migration from 0 or 1 actually performs the same
+		/// 0->1 was a noop as long as no values exceeded bounds
+		/// there is no problem if we enforce bounds again if the onchain_version is already v1
 		fn on_runtime_upgrade() -> Weight {
-			let weight = T::DbWeight::get().reads(1);
-			if StorageVersion::get::<Pallet<T>>() != 0 {
+			let current_version = Pallet::<T>::current_storage_version();
+			let onchain_version = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				target: TARGET,
+				"Running migration with current storage version {:?} / onchain {:?}",
+				current_version,
+				onchain_version
+			);
+
+			let mut translated = 0u64;
+			if onchain_version >= current_version {
 				log::warn!(
 					target: TARGET,
-					"skipping on_runtime_upgrade: executed on wrong storage version.\
-				Expected version 0"
+					"skipping on_runtime_upgrade: executed on wrong storage version."
 				);
-				return weight
+				return T::DbWeight::get().reads(1)
 			}
 
-			// we do not actually migrate any data, because it seems that the storage representation of Vec and BoundedVec is the same.
-			// as long as we check the bounds in pre_upgrade, we should be fine.
+			CommunityMetadata::<T>::translate::<UnboundedCommunityMetadata, _>(
+				|k: CommunityIdentifier, meta: UnboundedCommunityMetadata| {
+					info!(target: TARGET, "     Migrating community metadata for {:?}...", k);
+					translated.saturating_inc();
+					Some(meta.migrate_to_v2())
```

Comment:
ok, you are saying that this works for `v0` -> `v2` as well as `v1` -> `v2` because the storage representation of `UnboundedCommunityMetadata` and `CommunityMetadataV1` is the same?

Reply:
actually pretty sure it wont work for `v1` -> `v2` because `from_cropping` takes `Vec<u8>` and this will be the wrong type if the input is a `CommunityMetadataV1`.

Reply:
generic didn't work immediately, but actually not necessary. tested both cases now

Reply:
hmm ok. it seems counterintuitive to me but it probably works, because we somehow implicitly "cast" a`CommunityMetadataV1` into a `UnboundedCommunityMetadata`, because we operate direclty on the storage value..